app.js

```diff
@@ -10,84 +11,178 @@ app.use(express.static('public'));
+    .set('accept', 'json')
+    .set('Authorization', `Bearer ${bearerToken}`);
+
+/**
+ * this generates a signed jwt in order for us to receive an oauth access_token from getlabs
+ * NOTE: we sign the jwt with the apiToken as your secret
+ */
+const generateJwtForOauthToken = () => {
+    const jwt = require('jsonwebtoken');
+
+    // build header
+    const jwtHeader = {
+        algorithm: 'HS512',
+    };
+
+    // build payload
+    const jwtPayload = {
+        sub: getlabsConifg.clientId,
+        aud: `https://${getlabsConifg.hostname}/oauth/token`,
+    };
+
+    return jwt.sign(jwtPayload, getlabsConifg.apiToken, jwtHeader);
+};
+
+/**
+ * this method fetches an access token from getlabs api in order to make requests
+ * @type {Promise<unknown>}
+ */
+function setOauthTokenResp() {
+    return getHttpClient(generateJwtForOauthToken())
+        .post('/oauth/token')
+        .then(async (response) => {
+            console.log(`Token response received: storing token response`);
+            oauthTokenResp = response.body;
+        })
+        .catch((error) => {
+            console.error('Error fetching access token from Getlabs:', util.inspect(error.response?.body, false, null, true));
+            server.close();
+        });
+};
+
+/**
+ * this method is used to update your access token in the event that it expires
+ * @param refresh_token
+ */
+function refreshAccessToken(access_token, refresh_token) {
+    return getHttpClient(access_token)
+        .post(`/oauth/refresh_token`)
+        .query({ refresh_token: refresh_token})
+        .then((response) => {
+            console.log(`Refreshing tokens: success`);
+            oauthTokenResp = response.body;
+        })
+        .catch(async (error) => {
+            // if we get a 400 back then our refresh token is expired and we need to get a new one
+            if (error.status == 400) {
+                console.log('Refreshing tokens: failed')
+                await setOauthTokenResp();
+            } else {
+                console.error('Error fetching access token from Getlabs:', util.inspect(error.response?.body, false, null, true));
+            }
+        });
+
+}
 
 /**
- * Fetches or creates the patient during startup based on the loggedInUser set above.
+ * Function to Fetch or create the patient during startup based on the loggedInUser set above.
  * Stores the resulting promise so the the patient's ID can be added to requests.
  */
-const getlabsPatient = agent.post('/patient')
-  .send(loggedInUser)
-  .then((response) => {
-    console.log(`Patient fetched from https://${getlabsConifg.hostname}/patient, all API requests will be made for this patient:`, response.body);
-    return response.body;
-  })
-  .catch((error) => {
-    console.error('Error fetching patient from Getlabs:', util.inspect(error.response?.body, false, null, true));
-    server.close();
-  });
+function getGetlabsPatient(access_token) {
```

Comment:
This is no longer getting called at start up. It was a nice way of knowing if your user is correct when starting the app. Otherwise we need to wait until the payment step to see if there is an error.